### PR TITLE
[hyprctl] Expose the special workspace id and name of the monitor

### PR DESCRIPTION
**Pull request by 15cm**
So that we know if a special workspace is shown on a monitor

#### Describe your PR, what does it fix/add?
It adds an `specialWorkspace.id` field to the output of hyprctl.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
The added info allows me to workaround https://github.com/hyprwm/Hyprland/issues/2391 by writing a script to interact with hyprctl https://github.com/15cm/nixfiles/commit/561eb3a46d3f3a362e10dc35d4c8e0717ebc00a1

#### Is it ready for merging, or does it need work?
It's ready for merging. I've tried it locally with the hyprland binary on NixOS built on https://github.com/15cm/nixfiles/commit/561eb3a46d3f3a362e10dc35d4c8e0717ebc00a1. I didn't find a code format guideline in the repo. So I just format the code following the existing examples.

### CODE REVIEW

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -45,6 +45,9 @@ std::string monitorsRequest(HyprCtl::eHyprCtlOutputFormat format) {
         "id": %i,
         "name": "%s"
     },
+    "specialWorkspace": {
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:48]: why isnt this just a field

Comment (15cm) [reply src/debug/HyprCtl.cpp:55]: The field should have a `name` subfield in the future. I was not sure how to handle the special cases like `specialWorkspaceID == 0` so I didn't add it by myself.

I added a commit in this PR to include the name. PTAL. The output will be like:
```
    "specialWorkspace": {
        "id": -99,
        "name": "special"
    },
```
for a special workspace that's shown or a hidden special workspace like:
```
    "specialWorkspace": {
        "id": 0,
        "name": ""
    },
```

File: src/debug/HyprCtl.cpp [hunk 2]
```diff
@@ -19,6 +19,13 @@ static void trimTrailingComma(std::string& str) {
         str.pop_back();
 }
 
+static std::string getWorkspaceNameFromSpecialID(const int workspaceID) {
+    if(workspaceID == 0) return "";
```
[the remark this answers is not in the document]
Comment (15cm) [reply src/debug/HyprCtl.cpp]: Done.

[the remark this answers is not in the document]
Comment (15cm) [reply src/debug/HyprCtl.cpp]: `clang-format -i src/debug/HyprCtl.cpp` gives me

```
YAML:63:31: error: invalid boolean
AlignConsecutiveDeclarations: AcrossEmptyLines
                              ^~~~~~~~~~~~~~~~
Error reading <repo-path>.clang-format: Invalid argument
```

[the remark this answers is not in the document]
Comment (15cm) [reply src/debug/HyprCtl.cpp]: ok `nix-shell -p clang` provides a clang-format version 11. `nix-shell -p clang-tools` provides a clang-format version 14 that's high enough to recognize the new type of the option. Formatted.

**Review by vaxerski — APPROVED** [review]
gangsta

**Pull request merged**